packages/connectkit/src/components/Common/AmountInput/index.tsx

```diff
@@ -0,0 +1,365 @@
+import React, { useEffect, useState } from "react";
+import { ROUTES, usePayContext } from "../../DaimoPay";
+
+import {
+  ModalBody,
+  ModalContent,
+  PageContent,
+} from "../../Common/Modal/styles";
+
+import {
+  assertNotNull,
+  WalletBalanceOption,
+  WalletPaymentOption,
+} from "@daimo/common";
+import { AnimatePresence, motion } from "framer-motion";
+import { css } from "styled-components";
+import { formatUnits, parseUnits } from "viem";
+import { chainToLogo } from "../../../assets/chains";
+import styled from "../../../styles/styled";
+import {
+  formatUsd,
+  roundDecimals,
+  tokenAmountToFormattedUsd,
+  USD_DECIMALS,
+  usdToFormattedTokenAmount,
+} from "../../../utils/format";
+import { isValidNumber } from "../../../utils/validateInput";
+import Button from "../../Common/Button";
+import SwitchButton from "../../Common/SwitchButton";
+import CircleSpinner from "../../Spinners/CircleSpinner";
+import AmountInputField from "./AmountInputField";
+
+const MAX_USD_VALUE = 20000;
+
+const MultiCurrencySelectAmount: React.FC<{
+  selectedTokenBalance: WalletBalanceOption;
+  setSelectedTokenOption: (option: WalletPaymentOption) => void;
+  nextPage: ROUTES;
+}> = ({ selectedTokenBalance, setSelectedTokenOption, nextPage }) => {
+  const { paymentState, setRoute, triggerResize } = usePayContext();
+
+  const balanceToken = selectedTokenBalance.balance.token;
+  const isUsdStablecoin = balanceToken.fiatSymbol === "$";
+
+  const minimumMessage =
+    selectedTokenBalance.minimumRequired.usd > 0
+      ? `Minimum 
+  ${formatUsd(selectedTokenBalance.minimumRequired.usd, "up")}`
+      : null;
+
+  const [editableValue, setEditableValue] = useState("");
+  const [secondaryValue, setSecondaryValue] = useState(
+    usdToFormattedTokenAmount(0, selectedTokenBalance.balance.token),
+  );
+  const [isEditingUsd, setIsEditingUsd] = useState(true);
+  const [message, setMessage] = useState<string | null>(minimumMessage);
+  const [continueDisabled, setContinueDisabled] = useState(true);
+
+  useEffect(() => {
+    triggerResize();
+  }, [message]);
+
+  /**
+   * Update the editable value and secondary value, taking into account whether
+   * the user is currently editing the USD value or the token value.
+   */
+  const updateValues = (
+    newEditableValue: string,
+    isUsdValue: boolean,
+    newSecondaryValue?: string,
+  ) => {
+    const sanitizedEditableValue = /\d/.test(newEditableValue)
+      ? newEditableValue
+      : "0";
+
+    let usdValue: string;
+    let tokenValue: string;
+
+    if (isUsdValue) {
+      usdValue = roundDecimals(
+        Number(sanitizedEditableValue),
+        USD_DECIMALS,
+        "down",
+      );
```

Comment:
extract `parseUsd(sanitizedUsdStr)` , we are calling `roundDecimals(....)` copy+pasted